### DISCUSSION TITLE
Feature request: Add a "remove price limits" cheat

### DISCUSSION THREAD

**Discussion by MarcelVos96** [OpenRCT2/OpenRCT2#22616] [Feature requests]
The price cap in RollerCoaster Tycoon 2 for anything but the park entrance is 20 euros. In the past this could be circumvented by using some higher level method of setting the price, such as via a price manager plugin.

Recently negative prices were causing some issues, and as a result the prices were hard-limited at the bottom to €0.00 and at the top to €20.00, and even plugins cannot circumvent it anymore. However, this upper limit doesn't seem strictly necessary to me, and a cheat to increase this limit to something like €999 (same as park entrance) would both allow such plugins to charge fun ridiculous prices, and also allow you to do it yourself.

This "unlock price limits" cheat will work especially well in combination with the new "guests ignore prices" cheat and will allow for maximum tomfoolery.

**Comment by ZehMatt** [comment]
I don't think we should have yet another cheat. As for price limits, there should be none in my opinion as setting prices too high will already have negative effects so we don't need an arbitrary limit really. I think we should revert that change and just make sure its not negative.

**Comment by MarcelVos96** [comment]
That would also work. It would allow you to charge more than 20 bucks for umbrellas in the rain, making that even stronger, but other than that I don't see how this would increase exploits. And the umbrella exploit is already limited by how much money guests spawn with anyway.

You could also limit the "rain price" of umbrellas to 20 bucks to prevent this exploit, which I think might actually be a good solution. Vastly increase the price limit, but implement a new thing that guests will not pay anything more than 20 bucks for an umbrella in the rain.

**Reply by ZehMatt** [reply]
I think the higher the price of umbrellas the higher the chance they refuse to pay that much even when it rains, when its raining it should just increase the chance, so when the price is 20 we could say thats 2% chance of a purchase, if it rains then we could just turn it to 4%, that's just an example but that is what I would aim for.

**Reply by MarcelVos96** [reply]
I don't agree with this. The 20 dollar umbrella trick is very widely known, it's maybe even the most famous exploit in the entire game. If that exploit gets removed then so many people will not know why it suddenly doesn't work, as the majority of people don't keep up with OpenRCT2 updates.

I also don't really see a reason to remove it. It's been part of the game for 25 years and it hasn't caused any issues. If anything it's beneficial because when I do it in a video or livestream I often get comments on how funny it is that that works.

**Reply by ZehMatt** [reply]
Well it was just an example on how we could approach price limits without soft capping it.

**Reply by Spyrois2cool** [reply]
What if we made it so guests are less likely to buy umbrellas in the rain the higher above 20 it is. 20.10 could already make them 2% or 4% less likely, and the pattern continues until they just would...never buy one regardless. Percentages don't need to be that low, you could make them whatever you'd think would work. Would still allow for some hilarity while nerfing anything higher than 20 and keeping the age old funny exploit in there.

**Reply by Sadret** [reply]
I would support that. Keep the old behaviour until 20 GBP and gradually decrease the probability of buying above 20 GBP. Either linear decay (which would give a hard cap) or exponential decay.

**Comment by astraylife1** [comment]
At this point, why not just use the No Money cheat.

**Reply by MarcelVos96** [reply]
It's fun to make a lot of money, and if you make a ride that guests would pay 25 bucks for (not that uncommon with things like the Air Powered Vertical Coaster), it's fun to actually be able to charge that.

**Comment by Spyrois2cool** [comment]
I am all for this idea. I LOVED the ride price manager plugin being able to surpass that $20 limit. It kinda showed how good a ride's stats could get and proved that good stats could make for a really good profitable ride. But when they all get hard limited to 20, they don't stand out at all.

**Reply by SpartanFrederic104** [reply]
I support this idea too. Very often I would build multi-station Splash Boats rides with stats high enough that they could charge up to $31.20 for the first five months since opening, which would provide a massive boost to my park's profits. But now that ride prices are hard limited to $20, I no longer see the point in building extremely exciting rides with the potential to charge more than $20 even just for the first five months, even in scenarios where guests prefer higher intensity rides.

At this point the $20 cap should be removed (regardless of cheats or not) and the code should be modified to check that the price is not negative in order to avoid any undesirable effects.

**Comment by 733737** [comment]
I have discussed in the past my disdain for the $20 cap. Perhaps the limit could be 100 dollars, that way it is still within reason a peep may be able to afford the purchase.

**Comment by Harry-Hopkinson** [comment]
Maybe the "Guests ignore Prices" cheat could also remove the cap on the prime limits? That way it could serve a dual purpose.

**Reply by 733737** [reply]
I agree with ZehMatt that it makes no sense to lock higher prices than $20 behind cheats.
